### PR TITLE
chore: upgrade gcp-metadata to 0.4

### PR DESCRIPTION
**Pull request by kjin**
Supercedes #581

### CODE REVIEW

**Comment by kjin** [comment]
AppVeyor is green: https://ci.appveyor.com/project/matthewloring/cloud-trace-nodejs/build/1.0.1839

I believe the check didn't show up because NuGet was down for a large portion of the day today.

@GoogleCloudPlatform/node-team PTAL

File: .travis.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ env:
   - CXX=g++-4.8 GCLOUD_TRACE_NEW_CONTEXT=1
 
 script:
+  - if [ $TRAVIS_NODE_VERSION -ge 8 ]; then yarn list; fi
```
Comment (ofrobots) [inline .travis.yml]: Why is this needed?

Comment (kjin) [reply .travis.yml]: We don't print package structure for Node 8 tests so I wanted to add that for debuggability -- but I realized that we don't for any Node version tests.

I'll remove the conditional in an upcoming PR... don't want to wait for CI to pass again before landing this change.
